### PR TITLE
Support SSL termination

### PR DESCRIPTION
**Pull request by pazaan**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -14,7 +16,16 @@ const datatoworkbook = require('@tidepool/data-tools/bin/datatoworkbook');
 
 const log = logMaker('app.js');
 
-const port = 3001;
+const config = {};
+config.httpPort = process.env.HTTP_PORT;
+config.httpsPort = process.env.HTTPS_PORT;
+if (process.env.HTTPS_CONFIG) {
```
Comment (jh-bate) [inline app.js]: so unless I have read this wrong in order to setup as `https` we need both the `process.env.HTTPS_PORT` and `process.env.HTTPS_CONFIG` to be set??  In which case with the setup we have here we could be in the situation that they aren't but we still proceed ... see line 198-199

Comment (pazaan) [reply app.js]: Good point. Updated code to sort this out.

File: app.js [hunk 2]
```diff
@@ -53,6 +67,23 @@ app.use(bodyParser.urlencoded({
...
+    } else {
+      log.info('Redirecting HTTP request to HTTPS');
+      const httpsHost = req.headers.host.replace(/:\d+/, `:${config.httpsPort}`);
+      res.redirect(`https://${httpsHost}${req.url}`);
+    }
+  });
+}
+
+// The Health Check
+app.use('/status', require('express-healthcheck')());
```
Comment (jh-bate) [inline app.js]: Just as an FYI we don't have a consistent `status` endpoint response. 

Some, e.g. seagull in this case, return an indication of if crud handler is operational https://github.com/tidepool-org/seagull/blob/master/lib/routes/seagullApi.js#L204

And the newer platform code uses this middleware  https://github.com/ant0ine/go-json-rest#status

Comment (pazaan) [reply app.js]: Yeah, I discussed this with Darin. We'll talk about standardising this a bit more once it does something more meaningful ;)

**Pull request merged**